scripts/asm-installer/asm_vm

```diff
@@ -341,6 +342,15 @@ OPTIONS:
                                                       Required if
                                                       --service_account is
                                                       passed.
+  -m|--image_family           <IMAGE_FAMILY>          (optional) The image family
+                                                      for the operating system
+                                                      Only valid when the
+                                                      --source_instance_template
+                                                      is not specified. Default
```

Comment:
Why is this only valid when --source_instance_template is not specified? 

For example, when creating a source instance template for bookinfo, that template only specifies the logic for bootstrapping the mysql app. Maybe a better way is to allow source_instance_template and overwrite the image type in the source instance template? @jasonwzm WDYT?

Reply:
IIUC, this command should work. Otherwise it breaks CUJ
```
./asm_vm create_gce_instance_template \
asm-mysql-instance-template \
--project_id PROJECT_ID \
--cluster_location CLUSTER_LOCATION \
--cluster_name CLUSTER_NAME \
--workload_name mysql \
--image_family debian-10 \
--source_instance_template mysql-instance-template
```

Reply:
source template already include a linux distros https://cloud.google.com/sdk/gcloud/reference/compute/instance-templates/create

Reply:
I think we need to be clear on what this flag really entails. Does it mean the OS the service proxy agent supports or the OS that will be created/overridden in the source instance template? What if the user uses a custom image which is built on top of centos and they would like a service proxy agent for centos?

Reply:
What will happen if customer run command like this? 
```
./asm_vm create_gce_instance_template \ asm-mysql-instance-template \ --project_id PROJECT_ID \ --cluster_location CLUSTER_LOCATION \ --cluster_name CLUSTER_NAME \ --workload_name mysql \ --image_family debian-10 \ --source_instance_template mysql-instance-template
```

By looking at the script, it generates a message at `warn` level, what else does the script handle this situation?

Reply:
yes it will warn the user and the final instance-template linux distro will use the distro from mysql-instance-template